Convert Match student component to Angular

geoffreykwan opened the pull request:
You will need to run ```npm i``` locally and ```docker exec -it wise-client npm i``` in Docker container.

@breity Can you fix the styling in the Match student view?

Things to test
- regular text choice and buckets
- choice with an image
- bucket with an image
- student created choices
- student private notes as choices
- left/right buckets
- correct answer
- correct answer with correct order
- feedback
- correct feedback
- incorrect position feedback
- import work from another Match component
- show work from another Match component

Closes #161

geoffreykwan committed:
Convert Match student component to Angular. #161

geoffreykwan committed:
Ran extract-i18n for Match student. #161

geoffreykwan committed:
Merge branch 'develop' into issue-161-convert-match-student

geoffreykwan committed:
Add the option to copy choices and buckets when authoring a Match connected component. Fix a null pointer error when the student submits with a student created choice in a Match component. #161

geoffreykwan committed:
Merge branch 'develop' into issue-161-convert-match-student

geoffreykwan committed:
Remove old dragula library from package.json. Add reload preview button in Match authoring preview. #161

geoffreykwan committed:
Remove fdescribe from match-student.component.spec.ts that was accidentally left in. #161

geoffreykwan committed:
Remove old angular-dragula import. #161

hirokiterashima committed:
Broke up match-student into sub components. #161

hirokiterashima committed:
Extracted MatchStudentModule and StudentComponentModule. #161

hirokiterashima committed:
Moved DeleteChoice functionality into its own component. #161

hirokiterashima committed:
Cleaned up code. #161

hirokiterashima committed:
Reverted createdSourceBucket to return empty items. #161

hirokiterashima committed:
Replaced Dragula with Angular Material Drag and Drop. #161

hirokiterashima committed:
Revert "Replaced Dragula with Angular Material Drag and Drop. #161"

This reverts commit 96ca853bce20a5c6644a69289393232a0ded29ef.

hirokiterashima committed:
Merge branch 'develop' into issue-161-convert-match-student

hirokiterashima commented:
I reverted work to replace Dragula with Angular Material D&D. We will do that in a separate issue.

@breity I think this is ready for styling.

breity committed:
Updated layout and styles for Match student component.
Includes updates to Match choice item, add choice button, delete choice button components.
#161

breity committed:
Made Match orientation authoring text  RTL/LTR agnostic. #161

breity committed:
Updated Match student feedback section styling. #161

breity committed:
Do not add new Match choice on cancel.
Also updated some styles and i18n string.
#161

breity committed:
Show non-position feedback when choice does not have position-specific feedback. #161

breity committed:
Extracted Match status icon its own component. #161

breity committed:
Added missing input to ComponentSaveSubmitButtons.
Fixed disabling Submit button in Match student component.
#161

breity committed:
Updated AddMatchChoiceDialog to use FormBuilder.
Added localize string.
#161

breity committed:
Standardizedsome component content styles, layout. #161

breity committed:
Merge branch 'develop' into issue-161-convert-match-student

breity reviewed (approved):
Looking good. I updated styles and layout, fixed some bugs, made changes we discussed.

@hirokiterashima and @geoffreykwan, take a look and make sure everything is still working for you.

breity committed:
Added styling for notes and images in Match student component. #161

breity committed:
Remove status property from bucket items before saving. #161

breity committed:
Styled Match items for disabled components.
Updated i18n strings.

hirokiterashima reviewed (approved):
Looks good.

breity committed:
Fixed padding on Match items. #161
